client/Components/SingleCohort/SingleCohort.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react'
+import { connect } from 'react-redux'
+import { fetchCohort, fetchStudents, fetchStretches } from '../../store/thunks'
+import Sidebar from '../Sidebar'
+
+const mapStateToProps = ({ cohort, students, stretches }) => {
+  return {
+    cohort,
+    students,
+    stretches
+  }
+}
+
+const mapDispatchToProps = dispatch => {
+  return {
+    fetchCohort: cohortId => dispatch(fetchCohort(cohortId)),
+    fetchStudents: cohortId => dispatch(fetchStudents(cohortId)),
+    fetchStretches: cohortId => dispatch(fetchStretches(cohortId))
+  }
+}
+
+const SingleCohort = ({
+  match,
+  cohort,
+  students,
+  stretches,
+  fetchCohort,
+  fetchStudents,
+  fetchStretches
+}) => {
+  const cohortId = match.url.slice(8)
+  useEffect(() => {
+    fetchCohort(cohortId)
+    fetchStudents(cohortId)
+    fetchStretches(cohortId)
+  }, [])
+  return (
```

Comment:
I could definitely be wrong here since my knowledge of useEffect is an one-minute React article but since these are asynchronous, would you need to do "reutrn Promise.all([fetchCohort(cohortId),...])"?

Reply:
I don't think so? Let's chat about this tonight.

Reply:
Yea I'm quite sure you are right. I think I just said that b/c I unnecesarilly do it in my ComponentDidUpdates